File: List1/Ex1/python/lifo.py
class Stack:
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        if self.is_empty():
            print("Error: Trying to pop from an empty stack!")
            return None
        return self.items.pop()

    def peek(self):
        if self.is_empty():
            print("Error: Stack is empty!")
            return None
        return self.items[-1]

    def is_empty(self):
        return len(self.items) == 0

File: List1/Ex1/rust/src/lifo.rs
pub struct Stack<T> {
    items: Vec<T>,
}

impl<T> Stack<T> {
    // Tworzenie nowego stosu
    pub fn new() -> Self {
        Stack { items: Vec::new() }
    }

    // Dodawanie elementu do stosu
    pub fn push(&mut self, item: T) {
        self.items.push(item);
    }

    // Usuwanie elementu ze stosu
    pub fn pop(&mut self) -> Option<T> {
        if self.is_empty() {
            print!("Error: Trying to pop from an empty stack!");
            None
        } else {
            self.items.pop()
        }
    }

    // Sprawdzanie, czy stos jest pusty
    pub fn is_empty(&self) -> bool {
        self.items.is_empty()
    }
}
